frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/index.tsx

```diff
@@ -151,10 +165,17 @@ export default function SaveExport({
 
     setDownloadStatus({ isLoading: true, blur: true });
 
-    const heatmapNode = document.getElementById("view") as HTMLDivElement;
+    const heatmapNode = document.getElementById(
+      "heatmap-container-id"
+    ) as HTMLDivElement;
+    const viewNode = document.getElementById("view") as HTMLDivElement;
 
     // Options for the observer (which mutations to observe)
-    const config = { childList: true, subtree: true };
+    const config: MutationObserverInit = {
+      childList: true,
+      subtree: true,
+      attributes: true,
```

Comment:
we're listening for changes in attributes now so that this line:
`heatmapNode.classList.add("is-downloading");`

triggers the download without having the set the echarts renderer for all filetypes.